controllers/vaultsecret_controller.go

```diff
@@ -93,6 +93,24 @@ func (r *VaultSecretReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	ctx = context.WithValue(ctx, "kubeClient", r.GetClient())
 	ctx = context.WithValue(ctx, "restConfig", r.GetRestConfig())
 
+	if util.IsBeingDeleted(instance) {
```

Comment:
so we had an issue with finalizers also in vault secrets?

Reply:
yes - tested both randomsecret and vaultsecret and found them both to not delete correctly